tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -0,0 +1,201 @@
+#define MAX_DATA_SIZE 1000
+
+/* Implementation of safe malloc which returns NULL if the reuested size is 0.*/
+void *safeMalloc(size_t xWantedSize) {
+  if(xWantedSize == 0) {
```

Comment:
I think that "malloc(0)" is implementation defined.  You can get either NULL or an address.  Do we know that this consistent with the implementations that FreeRTOS is targeting?

Reply:
We are using safeMalloc in all our proofs now. It returns NULL if the requested size is 0.